rfd/0084-azure-backend.md

```diff
@@ -0,0 +1,130 @@
+
+### Configuration
+
+This are some options on how the configuration side might look like.
+
+The original SQL Backend RFD imagined something like:
+
+```yaml
+teleport:
+  storage:
+    type: postgres
+    addr: pgservername.postgres.database.azure.com
+    database: dbname_defaulting_to_teleport
+    tls:
+      ca_file: path/to/azure_postgres_trust_roots.pem
+    azure:
+      username: pgusername@pgservername
+```
+
+Alternatively, we could provide additional aliases for the Postgres backend:
+
+```yaml
+teleport:
+  storage:
+    type: azurepostgres
+    addr: pgservername.postgres.database.azure.com
+    username: pgusername@pgservername
+    database: dbname_defaulting_to_teleport
+    tls:
+      ca_file: path/to/azure_postgres_trust_roots.pem
+```
+
+Or we could add some "authentication type" field:
+
+```yaml
+teleport:
+  storage:
+    type: postgres
+    addr: pgservername.postgres.database.azure.com
+    username: pgusername@pgservername
+    database: dbname_defaulting_to_teleport
+    authentication: azure
+    tls:
+      ca_file: path/to/azure_postgres_trust_roots.pem
+```
+
+Other options such as checking for the `.postgres.database.azure.com` suffix in the hostname seem a bit too fragile.
+
+Authentication will happen according to the default behavior of [the Azure Go SDK](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity), which will try to use credentials from envvars, from the IDMS accessible from Azure VMs, and from the command-line `az` tool - it would be straightforward to add some tunables to the configuration file if some specific need arose in the future.
+
+## Audit log
+
+The current two cloud storage options (DynamoDB on AWS and Firestore on GCP) can store both the auth backend and the events in the audit log. For user convenience, we'll implement audit log storage for the `sqlbk`/Postgres backend - an alternative would be to store the audit log in Cosmos DB, but that would require that our users set up both a Postgres server and a Cosmos DB account. As a bonus, this storage option will also be usable outside of Azure.
+
+An `audit` table will be added to the current db schema, with indices on a `time` field and an optional `sid` field.
+
+Assuming it's fine to store the log in the same database as the backend, a minimal configuration could be as simple as:
+
+```yaml
+teleport:
+  storage:
+    ...
+
+    audit_events_uri: ['backend://']
```

Comment:
What does `backend://` mean if your using a non-Postgres backend?

Reply:
That's going to error during initialization with something like "backend foo doesn't support audit log storage".

The implementation is going to be an optionally implemented `IntegratedAuditLog() IAuditLog` method that can return nil (so that it can pass through backend wrappers like Reporter and Sanitizer, I think.

Reply:
Removed this whole concept, actually - the implementation would be much cleaner from a blank slate rather than piggybacking onto the backend, anyway.